functions/src/middlewares/auth.js

```diff
@@ -20,6 +21,54 @@ const FBauth = async (req, res, next) => {
         return res.status(FORBIDDEN).json({ message: error.message, status: 'error' });
     }
 };
+
+const getAuthToken = (req, res, next) => {
+    if (
+        req.headers.authorization &&
+      req.headers.authorization.split(' ')[0] === 'Bearer'
+    ) {
+        req.authToken = req.headers.authorization.split(' ')[1];
+    } else {
+        req.authToken = null;
+    }
+    next();
+};
+  
+const checkIfAuthenticated = (req, res, next) => {
+    getAuthToken(req, res, async () => {
+        try {
+            const { authToken } = req;
+            const userInfo = await admin
+                .auth()
+                .verifyIdToken(authToken);
+            req.authId = userInfo.uid;
+            return next();
+        } catch (e) {
+            return res
```

Comment:
I have already written a middleware to check for authentication and authorization, all we have to do is call it as a middleware. The FBAuth there stands for firebase authenication.